### ISSUE TITLE
Setting up Cluster

### ISSUE THREAD

**Issue by tuxpowered** [dOpensource/dsiprouter#437]
I have been happy with dSIP, but have found the documentation extremely lacking in details or completely omitted for features.
I am happy to contribute documentation, as I uncover things to help improve the quality. 

I have searched and searched and have not been able to find any documentation on how to properly configure dSIPRouter to operate as a cluster. Although I see support in the `dsiprouter.sh` script my attempts at playing around have proved to be faulty.

I have taken a working dSIP install, and issued the following:

`dsiprouter clusterinstall root:password@10.10.10.51 -- -all -servernat --dsip-clusterid=100 --dsip-clustersync=100`

And while this did perform a remote installation, including a database on the remote server, it did not setup sync between them. (e.g.: making a change such as adding an endpoint on node1 (10.10.10.50) does not appear to get sent to node2. 

Further, it seems that the web logins on node2 are not the same as node1.  It would seem that if the nodes were in a cluster, (active-active) as the documentation states, any changes to node1 would be reflected in node2, thus it would be expected that login credentials would have synced from node1 to node2 on the cluster install. 

I am left with the conclusion that one of 2 things is happening here.

1. I am issuing the wrong commands to set up the proper cluster, _due to a lack of documentation_. 
2. The lack of documentation is intentional void of the information which I can only assume would be that  @mackhendricks does not wish to make the information publicly available. _(I truly hope this is not the case)_

As I started off by stating I am happy to make updates to documentation as I work thru the missing parts. But sometimes you just need to see how it works before you can properly document something :)

If anyone has set up the cluster and cares to share, again I am happy to draft up docs for the project.

**Comment by mackhendricks** [comment]
Yeah...we haven't put time and effort into making cluster support in dSIPRouter core easy to use.  Let us know if you want to contribute some time to making the cluster portion easy to use.    We really need someone to own that piece.  We currently implement clustering for customers under paid engagements because the command line or the UI doesn't make it easy.

**Comment by tuxpowered** [comment]
@mackhendricks Thank you for clarifying that for everyone here. 
I would be happy to help out by making the clustering easier, perhaps with a dedicated configuration script. 
Would you happen to be able to pass over some setup guidelines? (privately if need be)? That would serve as a starting point for me to start documenting and performing system checks and tests to start crafting an easy-to-manage setup for users. 

Perhaps after that is done I can make a blueprint for flask to let setup from the GUI, or at least the ability to 'join' an existing cluster. (that might be better).

On the note of Flask, I noticed that dSIPRouter is based on a very old version of Flask. I am curious why it seems there is no move to the 2. x branch, esp. considering there have been about 10 releases from the current version.  As python2.x is `dead` it would be good to be migrating to the current versions of the flask and python3.9+ 

if you are able to provide some starting guides to setup, let me know the best way to connect with you. 
Thank you again @mackhendricks for the response.

**Comment by mackhendricks** [comment]
I will be working on roadmap this week and I will update this issue.   Also, we are moving dSIP to GoLang.  We want the install of dSIP to be one binary.

**Comment by tuxpowered** [comment]
That would be helpful. Moving to goLang... shame I do not know GO as well as python. But I was kinda hoping maybe a move to FastAPI :) 
_Automatic API documentation, async, etc._ 

is there a public test/dev build in golang ?

**Comment by devopsec** [comment]
the install issues were resolved in:
- ca9e5da8c80b3f9fb09752ebd8e1d2662a55f59a
- 997f911ecce2f4df036a30bed2e488ca983c3fba
- 6d2aa297487b38829ba832aec10ddc1e3b3d7f10
- ccea19047f2b1959fe0bbc0e6cf70a66576e5d15

the documentation could use some updating though
